src/tests.c

```diff
@@ -400,6 +400,16 @@ void run_scratch_tests(void) {
     secp256k1_scratch_space_destroy(none, scratch);
     CHECK(ecount == 5);
 
+    /* Test that large integers do not wrap around in a bad way */
+    scratch = secp256k1_scratch_space_create(none, 1000);
+    /* try max allocation with a large number of objects */
+    CHECK(!secp256k1_scratch_max_allocation(&none->error_callback, scratch, (SIZE_MAX / (ALIGNMENT - 1)) + 1));
```

Comment:
`ALIGNMENT` could be 1 here too, yielding UB the tests.

Reply:
fixed